packages/core/strapi/lib/types/strapi.d.ts

```diff
@@ -0,0 +1,291 @@
+import type Koa from 'koa';
+
+import type { StringMap } from './utils';
+
+type Controller = {
+  [methodName: string | number | symbol]: (context: Koa.Context) => unknown;
+}
+
+/**
+ * The Strapi interface implemented by the main Strapi class.
+ */
+export interface Strapi {
+  /**
+   * Getter for the Strapi enterprise edition configuration
+   */
+  readonly EE: unknown;
+
+  /**
+   * Getter for the Strapi configuration container
+   */
+  readonly config: unknown;
+
+  /**
+   * Getter for the Strapi auth container
+   */
+  readonly auth: unknown;
+
+  /**
+   * Getter for the Strapi sanitizers container
+   */
+  readonly sanitizers: unknown;
```

Comment:
we have some problems here, if the type is unknown, for example if the user tries this `strapi.sanitizers.get()` we'll have this error "Property 'get' does not exist on type 'unknown'.".

Same with any property that is typed as unknown when it's an object with methods, I know you said we want to type more layers later in the future, but this makes those functions unusable in a typescript project

Reply:
@Convly

Reply:
My vision on this was to force the user to cast used methods to the desired type right away (even `any` if needed).
We could decide to use any instead of `unknown` but it'll be less strict & peoples.
Might be a good idea to have other opinions on the matter

Reply:
Not sure what the best approach will be, but as unkown that means we are forcing the users to type it themselves basically (they might choose to just put any).
But since we are planning to type everything, what's the harm in putting it as 'any' ? this is going to change later anyways

Reply:
I don't have much context on this, but if you plan to type those registries later then yes `any` would be best suited, I think it's better to opt-out from type checking than to provide a type that doesn't reflect the full truth.

As those registries seem to share a common backbone (a `get` function as mentioned) you can later add a templated interface for them to extend once you're ready to switch away from `any`.

Reply:
Alright, thanks for your input, I'll proceed with any while we wait for more in-depth typings.